src/protocols/WebSocket.ts

```diff
@@ -83,8 +85,17 @@ export default class WebSocketProtocol extends BaseProtocolRealtime {
 
       this.client = new this.WebSocketClient(url, this.options);
 
+      this.client.onping = () => {
+        this.heartbeat(this.client);
+      };
+
+      this.client.onpong = () => {
+        return resolve();
+      };
```

Comment:
1. you shouldn't resolve the promise in `onpong`: you're resolving the promise multiple times
2. your 'onping' method will only work if kuzzle does send PINGs, but that may very well not be the case, as Kuzzle can be configured to have passive checks (i.e. it will responds to PINGs with PONGs, but it won't send PING packets by itself), see here: https://github.com/kuzzleio/kuzzle/blob/master/.kuzzlerc.sample#L419
3. onping/onpong methods don't exist in the browser implementation of websockets, you'll have to create PING packets manually, and check if a received packet is a pong one (see https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_servers#pings_and_pongs_the_heartbeat_of_websockets)

What you should do instead is to manually send PING packets to Kuzzle every `pingInterval` milliseconds, and then check received messages to verify if it's a PONG response, which resets the heartbeat timeout.

So you shouldn't have "onping"/"onpong" methods at all.